core/src/main/java/io/questdb/std/IntLongPriorityQueue.java

```diff
@@ -122,11 +109,11 @@ else if (midVal > v)
                 return mid;
             }
         }
-        return low;
+        return scanSearch(v, low);
```

Comment:
Shouldn't we pass the `high` value in this call? What if we have a 1M entry input and the searched number is located in the first few entries? In this case, we'll end up scanning almost 1M entries. Am I missing anything?

Reply:
Hm...why? We know that result contained in the [low..high) range due to binary search invariant. So, if we will scan original array starting from the low - we must eventually hit the target (in this case - value strictly greater than `v`) and perform no more than `high - low` steps.

Reply:
My bad. Yes, thanks to the previous checks and the order invariant, `high` value is not required.